src/lib.rs

```diff
@@ -48,25 +44,36 @@ impl LinkDrop {
     }
 
     /// Create new account and and claim tokens to it.
-    pub fn create_account_and_claim(&mut self, new_account_id: AccountId, new_public_key: String) -> Promise {
-        assert_eq!(env::signer_account_id(), env::current_account_id());
+    pub fn create_account_and_claim(&mut self, new_account_id: AccountId, new_public_key: Base58PublicKey) -> Promise {
+        assert_eq!(env::signer_account_id(), env::current_account_id(), "Create account and claim only can come from this account");
+        assert!(env::is_valid_account_id(new_account_id.as_bytes()), "Invalid account id");
         let amount = self
             .accounts
             .remove(&env::signer_account_pk())
             .expect("Unexpected public key");
         Promise::new(env::current_account_id()).delete_key(env::signer_account_pk());
         Promise::new(new_account_id)
```

Comment:
The only problem with this function is `new_account_id` might already exist, in this case, the transaction will fail and the money will be lost (returned back to this contract).

Ideally you should attach a callback to itself which will check that the promise succeeded and then delete the access key. But if the promise has failed, it should credit the amount back to the internal public key account.

Reply:
Or if the `new_account_id` is too short, it will do the same.